share/mk/src.opts.mk

```diff
@@ -428,7 +428,7 @@ __DEFAULT_NO_OPTIONS+=PIE
 #    !${MACHINE_CPU:Mcheri} || ${MACHINE_ABI:Mpurecap}
 # but that logic doesn't work in Makefile.inc1...
 .if (${__C} != "cheri" && ${__C} != "morello") || \
-    (${__T:Mmips64*c*} || ${__T:Mriscv64*c*} || ${__T:Mmorello})
+    (${__T:Mmips64*c*} || ${__T:Mriscv64*c*} || ${__T:Maarch64*c*})
```

Comment:
Sorting these would be nicer, but we can do that later as part of pulling things into dev (and this commit will need to be folded into previous ones anyway to ensure that the history has TARGET_ARCH as always being aarch64c).